src/System.IO.FileSystem/src/System/IO/Win32FileSystemObject.cs

```diff
@@ -64,7 +64,7 @@ public DateTimeOffset CreationTime
                 {
                     EnsureDataInitialized();
                     long dt = ((long)(_data.ftCreationTimeHigh) << 32) | ((long)_data.ftCreationTimeLow);
-                    return DateTimeOffset.FromFileTime(dt);
+                    return DateTimeOffset.FromFileTime(dt).ToLocalTime();
```

Comment:
The windows calls to ToLocalTime() can be safely removed after all.

It looks like the fault can be further narrowed down to FromUnixTimeSeconds/FromUnixTimeMilliseconds in DateTimeOffset.cs in mscorlib; they return a Zero Timespan when they should be returning Timespan representing the offset of the time zone from UTC.

DateTime has a FromFileTime that is the same as the one in DateTimeOffset. Would we also want a FromUnixTime*seconds in DateTime and have the DateTimeOffset delegate to it?